### PR TITLE
many: replace interfaces.Repository.AddSnap with AddAppSet

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This change will enable us to inject an app set into the interfaces.Connected(Plug|Slot) types, which will allow interfaces to correctly generate label expressions for connected plugs/slots.

### CODE REVIEW

File: snap/component.go [hunk 1]
```diff
@@ -41,17 +41,10 @@ type ComponentInfo struct {
...
-	return &ComponentInfo{
-		Component:   cref,
-		Type:        ctype,
-		Version:     version,
-		Summary:     summary,
-		Description: description,
-	}
+	// ComponentSideInfo contains information for which the source of truth is
+	// not the component blob itself.
+	ComponentSideInfo
```
Comment (alfonsosanchezbeato) [inline snap/component.go:47]: Why is it needed now? It becomes accessible from SnapAppSet (rev)

File: interfaces/repo.go [hunk 2]
```diff
@@ -1000,10 +953,12 @@ func (r *Repository) AddSnap(snapInfo *snap.Info) error {
 
 	snapName := snapInfo.InstanceName()
 
-	if r.plugs[snapName] != nil || r.slots[snapName] != nil {
+	if r.appSets[snapName] != nil || r.plugs[snapName] != nil || r.slots[snapName] != nil {
```
Comment (alfonsosanchezbeato) [inline interfaces/repo.go:956]: Should be enough with checking `r.appSets[snapName]` I guess?

File: snap/component.go (hunk 1 above)
Comment (alfonsosanchezbeato) [inline snap/component.go:47]: Why do we need ComponentSideInfo now? I see that in the end you will have access to the revision from a SnapAppSet, but not sure if I see the use case in this PR, is that maybe in one of the follow-ups? Also, not sure why the revision would be needed from the interfaces code.

**Review by alfonsosanchezbeato — COMMENTED** [review]
Couple of comments around

File: snap/component.go [hunk 3]
```diff
@@ -41,17 +41,10 @@ type ComponentInfo struct {
 	// component is associated with. This field may be empty if the
 	// ComponentInfo was not created with the help of a snap.Info.
 	Hooks map[string]*HookInfo `yaml:"-"`
-}
 
-// NewComponentInfo creates a new ComponentInfo.
-func NewComponentInfo(cref naming.ComponentRef, ctype ComponentType, version, summary, description string) *ComponentInfo {
```
[the remark this answers is not in the document]
Comment (andrewphelpsj) [reply snap/component.go:47]: This is used much later in “snap run” to set some environs variables

File: interfaces/repo.go [hunk 4]
```diff
@@ -973,23 +923,26 @@ func (r *Repository) SnapSpecification(securitySystem SecuritySystem, appSet *Sn
 	return spec, nil
 }
 
-// AddSnap adds plugs and slots declared by the given snap to the repository.
+// AddAppSet adds plugs and slots declared by the given snap to the repository.
```
Comment (pedronis) [inline interfaces/repo.go:926]: this doc comment probably need a bit more updates, as the AppSet is not strictly just about the snap anymore

**Review by pedronis — COMMENTED** [review]
lgtm, doc comment question

**Comment by andrewphelpsj** [comment]
@pedronis @alfonsosanchezbeato I pushed a new commit (a37f618ee565b13f95969506cd0348bffc51baed) that fixes a bug wrt implicit slots.

**Comment by alfonsosanchezbeato** [comment]
> @pedronis @alfonsosanchezbeato I pushed a new commit ([a37f618](https://github.com/snapcore/snapd/commit/a37f618ee565b13f95969506cd0348bffc51baed)) that fixes a bug wrt implicit slots.

Looks fine to me, thanks for the explanation in mm

**Pull request merged**